sdk/core/core-client-rest/src/multipart.ts

```diff
@@ -0,0 +1,200 @@
+      return descriptor.headers[actualHeaderName];
+    }
+  }
+
+  return undefined;
+}
+
+function getPartContentType(descriptor: PartDescriptor): HeaderValue | undefined {
+  const contentTypeHeader = getHeaderValue(descriptor, "content-type");
+  if (contentTypeHeader) {
+    return contentTypeHeader;
+  }
+
+  // Special value of null means content type is to be omitted
+  if (descriptor.contentType === null) {
+    return undefined;
+  }
+
+  if (descriptor.contentType) {
+    return descriptor.contentType;
+  }
+
+  const { body } = descriptor;
+
+  if (body === null || body === undefined) {
+    return undefined;
+  }
+
+  if (typeof body === "string" || typeof body === "number" || typeof body === "boolean") {
+    return "text/plain; charset=UTF-8";
+  }
+
+  if (body instanceof Blob) {
+    return body.type || "application/octet-stream";
+  }
+
+  if (isBinaryBody(body)) {
+    return "application/octet-stream";
+  }
+
+  // arbitrary non-text object -> generic JSON content type by default. We will try to JSON.stringify the body.
+  return "application/json; charset=UTF-8";
+}
+
+/**
+ * Enclose value in quotes and escape special characters, for use in the Content-Disposition header
+ */
+function escapeDispositionField(value: string): string {
+  return JSON.stringify(value);
+}
+
+function getContentDisposition(descriptor: PartDescriptor): HeaderValue | undefined {
+  const contentDispositionHeader = getHeaderValue(descriptor, "content-disposition");
+  if (contentDispositionHeader) {
+    return contentDispositionHeader;
+  }
+
+  if (
+    descriptor.dispositionType === undefined &&
+    descriptor.name === undefined &&
+    descriptor.filename === undefined
+  ) {
+    return undefined;
+  }
+
+  const dispositionType = descriptor.dispositionType ?? "form-data";
+
+  let disposition = dispositionType;
+  if (descriptor.name) {
+    disposition += `; name=${escapeDispositionField(descriptor.name)}`;
+  }
+
+  const filename =
+    descriptor.filename ??
+    (typeof File !== "undefined" && descriptor.body instanceof File
+      ? descriptor.body.name || undefined
+      : undefined);
+
+  if (filename) {
+    disposition += `; filename=${escapeDispositionField(filename)}`;
+  }
+
+  return disposition;
+}
+
+function normalizeBody(body?: unknown, contentType?: HeaderValue): MultipartBodyType {
+  if (body === undefined) {
+    // zero-length body
+    return new Uint8Array([]);
+  }
+
+  // binary and primitives should go straight on the wire regardless of content type
+  if (isBinaryBody(body)) {
+    return body;
+  }
+  if (typeof body === "string" || typeof body === "number" || typeof body === "boolean") {
+    return stringToUint8Array(String(body), "utf-8");
+  }
+
+  // stringify objects for JSON-ish content types e.g. application/json, application/merge-patch+json, application/vnd.oci.manifest.v1+json, application.json; charset=UTF-8
+  if (contentType && /application\/(.+\+)?json(;.+)?/.test(String(contentType))) {
```

Comment:
case sensitive or not?

Reply:
Hmm, yeah, apparently MIME types are case-insensitive. I'll make this case-insensitive, but there are definitely other places where we're just doing a plain equality check on the content type... maybe we shouldn't be doing that?